ecaudit/src/main/java/com/ericsson/bss/cassandra/ecaudit/logger/ChronicleWriter.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright 2019 Telefonaktiebolaget LM Ericsson
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.ericsson.bss.cassandra.ecaudit.logger;
+
+import java.io.File;
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.BlockingQueue;
+import java.util.function.Function;
+
+import com.google.common.annotations.VisibleForTesting;
+
+import net.openhft.chronicle.queue.ChronicleQueue;
+import net.openhft.chronicle.queue.ChronicleQueueBuilder;
+import net.openhft.chronicle.queue.ExcerptAppender;
+import net.openhft.chronicle.queue.impl.single.SingleChronicleQueueBuilder;
+import net.openhft.chronicle.wire.WriteMarshallable;
+import org.apache.cassandra.concurrent.NamedThreadFactory;
+
+class ChronicleWriter implements AutoCloseable
+{
+    private static final int BUFFER_SIZE = 256;
+
+    private final Thread writerThread = new NamedThreadFactory("Chronicle Writer").newThread(this::writerLoop);
```

Comment:
TED: You could give a lambda or a method reference to newThread(), instead of this...
Then the class does not have to implement Runnable and I also think it makes the code more clear since you can give the method a "good name".
I think it is simpler to understand/follow a method reference compared to finding the run method and look into it to see what it is doing...

Reply:
Done